pkg/routeagent_driver/handlers/ovn/config_syncer.go

```diff
@@ -0,0 +1,148 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ovn
+
+import (
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/submariner/pkg/routeagent_driver/constants"
+	iptcommon "github.com/submariner-io/submariner/pkg/routeagent_driver/iptables"
+	"github.com/vishvananda/netlink"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+)
+
+func (ovn *Handler) startSyncer(stop chan struct{}) {
+	logger.Info("Starting route config syncer")
+	// Start monitoring routes
+	go wait.Until(func() {
+		err := ovn.monitorRoutingTable(stop)
+		if err != nil {
+			logger.Errorf(err, "error starting monitorRoutingTable for interface: %q", OVNK8sMgmntIntfName)
+		}
+	}, time.Second, stop)
+}
+
+func (ovn *Handler) monitorRoutingTable(stop chan struct{}) error {
+	iface, err := netlink.LinkByName(OVNK8sMgmntIntfName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find interface: %q", OVNK8sMgmntIntfName)
+	}
+
+	addrCh := make(chan netlink.AddrUpdate)
+	doneCh := make(chan struct{})
+
+	err = netlink.AddrSubscribe(addrCh, doneCh)
+	if err != nil {
+		return errors.Wrapf(err, "failed to subscribe to address updates")
+	}
+
+	defer close(doneCh)
+
+	var prevIP net.IP
+
+	assignPrevIP := func() {
+		if addrs, err := netlink.AddrList(iface, netlink.FAMILY_V4); err == nil && len(addrs) > 0 {
+			prevIP = addrs[0].IP
+		}
+	}
+
+	handleAddressChange := func(addrUpdate netlink.AddrUpdate) {
+		if addrUpdate.LinkIndex != iface.Attrs().Index {
+			return
+		}
+
+		if !addrUpdate.NewAddr {
+			// Address deleted
+			prevIP = nil
+			return
+		}
+
+		addrs, err := netlink.AddrList(iface, netlink.FAMILY_V4)
+		if err != nil || len(addrs) == 0 {
+			return
+		}
+
+		newIP := addrs[0].IP
+		// If hte new IP address is not nil and different from the previous IP address,
+		// which mean an IP address is assigned to the interface. With OVN pod restart
+		// it is first cleared and the assigned again.
```

Comment:
Instead of saying OVN pod, please use the OVN pod name.

```suggestion
		// which mean an IP address is assigned to the interface. When OVN pod restarts
		// the IP address is first cleared and then assigned again.
```